### ISSUE TITLE
Color inline SVG fill="currentColor" with CSS 

### ISSUE THREAD

**Issue by xgerber** [Kozea/WeasyPrint#1497]
Hello,

In the context of a project I automatically generate HTML files that are transformed into PDF. These HTML files contain SVG pictograms. For the moment the SVG are included inline in the PDF but I can't give a color with the CSS, any idea?

```html
<div class="block-container block-second-column block-second-column-third-elem">
            <div class="block simple-column-block block-fond block-group green">
                <div class="block-picto">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 28 28">
                        <path fill="currentColor" d="M9 24h10v2H9v-2Zm0-8h10v2H9v-2Zm0 4h10v2H9v-2ZM27.679 8.266l-5.417-5a1.004 1.004 0 0 0-.311-.196l-7.583-3a1.006 1.006 0 0 0-.736 0l-7.583 3c-.114.045-.22.112-.311.196l-5.417 5A.996.996 0 0 0 0 9v3a1 1 0 1 0 2 0V9.438L6.958 4.86 14 2.075l7.042 2.786L26 9.438V26h-5V15a1 1 0 0 0-1-1H8a1 1 0 0 0-1 1v11H5V11.18l3.743-3.802L14 5.09l5.257 2.287L23 11.18V23a1 1 0 1 0 2 0V10.77c0-.263-.104-.515-.287-.703l-4.167-4.23a.985.985 0 0 0-.314-.216l-5.833-2.538a.998.998 0 0 0-.798 0L7.768 5.62a.985.985 0 0 0-.314.216l-4.167 4.23c-.183.188-.287.44-.287.703V26H2V16a1 1 0 1 0-2 0v11a1 1 0 0 0 1 1h26a1 1 0 0 0 1-1V9a.996.996 0 0 0-.321-.734Z"/>
                    </svg>
                </div>
            </div>
        </div>
```

```css
.green {
    color: rgb(60, 178, 18);
}
```

Browser : 

![image](https://user-images.githubusercontent.com/14215532/141973341-a6728e89-1d1a-4704-a515-25df7c73216e.png)

PDF WeasyPrint : 

![image](https://user-images.githubusercontent.com/14215532/141973504-b4f2b1b1-a001-4d6c-87c5-6abd9e82d017.png)

**Comment by Stigjb** [comment]
The value `currentColor` is a special case in the `tinycss2` package [^1] which is used to parse color values, and it seems to be replaced with black in Weasyprint here [^2].

[^1]: https://github.com/Kozea/tinycss2/blob/be5ee803d96a9c29014df5e66ddab95811ece7f3/tinycss2/color3.py#L41
[^2]: https://github.com/Kozea/WeasyPrint/blob/9a720ac08181fec3f91da22b604fac7f594e8c09/weasyprint/svg/__init__.py#L154

**Comment by liZe** [comment]
> it seems to be replaced with black in Weasyprint

You’re right, it’s an obviously broken workaround and it should be fixed!

**Comment by michaeloliverx** [comment]
This feature would be useful as inline SVGs could coloured with CSS.

> > it seems to be replaced with black in Weasyprint
> 
> You’re right, it’s an obviously broken workaround and it should be fixed!

Is it as simple as removing these lines?
https://github.com/Kozea/WeasyPrint/blob/9a720ac08181fec3f91da22b604fac7f594e8c09/weasyprint/svg/__init__.py#L154-L155

**Comment by liZe** [comment]
> Is it as simple as removing these lines?

Unfortunately, it’s more complicated than that. CSS inheritance is really, really limited in our SVG renderer (as it was in CairoSVG), and supporting this feature correctly would require to rewrite the whole CSS management for SVG 😒.

**Comment by danwos** [comment]
As I don't have control over the SVG content (generated via Sphinx and its extension), I changed the lines as follows to get at least the build done:
https://github.com/Kozea/WeasyPrint/blob/e9c45c7060036a65b62dd05a944199be1f441ffd/weasyprint/svg/__init__.py#L619
to
```python
try:
    red, green, blue, alpha = color(stroke_color)
except ValueError:
    if stroke_color == "currentColor":
        red, green, blue, alpha = (0, 0, 0, 1)
```
I'm sure this is not helpful for each use case. But at least the build is done and I can test weasyprint and check my CSS settings.

**Comment by subhanshu-shukla-ril** [comment]
Do we have any update update in this, My coloured SVG becoming black

**Comment by liZe** [comment]
> Do we have any update update in this, My coloured SVG becoming black

@subhanshu-shukla-ril No, we usually add comments when we have news.
